app/src/main/java/cm/aptoide/pt/analytics/analytics/RealmEventMapper.java

```diff
@@ -0,0 +1,38 @@
+package cm.aptoide.pt.analytics.analytics;
+
+import cm.aptoide.pt.database.realm.RealmEvent;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by trinkes on 12/01/2018.
+ */
+
+public class RealmEventMapper {
+  private final ObjectMapper objectMapper;
+
+  public RealmEventMapper(ObjectMapper objectMapper) {
+    this.objectMapper = objectMapper;
+  }
+
+  public RealmEvent map(Event event) throws JsonProcessingException {
+    return new RealmEvent(event.getTimeStamp(), event.getEventName(), event.getAction()
+        .ordinal(), event.getContext(), objectMapper.writeValueAsString(event.getData()));
+  }
+
+  public List<Event> map(List<RealmEvent> realmEvents) throws IOException {
+    ArrayList<Event> events = new ArrayList<>();
+    for (RealmEvent realmEvent : realmEvents) {
+      events.add(new Event(realmEvent.getEventName(),
+          objectMapper.readValue(realmEvent.getData(), new TypeReference<Map<String, Object>>() {
```

Comment:
question: is it possible that the json is invalid in some weird case? we are writting with the mapper so I don't think it is possible but if it happens it will throw and error and the chain in AptoideBiAnalytics might enter a loop(not sure if it is possible aswell)?